src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(damage) {
+    if (damage instanceof Herbivore && damage.hidden === false) {
+      damage.health -= 50;
+    }
+
+    if (damage.health === 0) {
+      Animal.alive.splice(Animal.alive.indexOf(damage));
+    }
```

Comment:
anyway,  you delete the last animal if this animal already died. If you already deleted an animal from alive array, indexOf will return -1 and you will delete -1 element (last element)